FalloutScriptDecompile.cpp

```diff
@@ -539,22 +539,25 @@ uint32_t CFalloutScript::BuildTreeBranch(CNodeArray& NodeArray, uint32_t nStartI
         if (wOperator == COpcode::O_IF)
         {
             // process possible conditional expression - this may be either normal IF statement or (x IF y ELSE z) expression
-            uint32_t ulElseOffset = NodeArray[j].m_Arguments[0].m_Opcode.GetArgument();
+            CNodeArray& arguments = NodeArray[j].m_Arguments;
+            uint32_t ulElseOffset = arguments[0].m_Opcode.GetArgument();
             int32_t ulElseIndex, ulSkipIndex = -1;
             ulElseIndex = BuildTreeBranch(NodeArray, j + 1, ulElseOffset); // true branch
-            if (NodeArray[ulElseIndex - 1].m_Opcode.GetOperator() == COpcode::O_JMP)
+            CNode& jumpNode = NodeArray[ulElseIndex - 1];
+            if (jumpNode.m_Opcode.GetOperator() == COpcode::O_JMP)
             {
-                uint32_t ulSkipOffset = NodeArray[ulElseIndex - 1].m_Opcode.GetArgument();
+                uint32_t ulSkipOffset = jumpNode.m_Opcode.GetArgument();
                 if (ulSkipOffset > NodeArray[j].m_ulOffset)
                 {
                     ulSkipIndex = BuildTreeBranch(NodeArray, ulElseIndex, ulSkipOffset); // false branch
-                    if (NodeArray[ulElseIndex - 2].IsExpression() && NodeArray[ulSkipIndex - 1].IsExpression())
+                    if (ulElseIndex == j + 3 && ulSkipIndex == ulElseIndex + 1 &&

```

Comment:
`FalloutScriptDecompile.cpp:553:37: warning: comparison of integer expressions of different signedness: ‘int32_t’ {aka ‘int’} and ‘uint32_t’ {aka ‘unsigned int’} [-Wsign-compare]`

Reply:
Interesting why VS doesn't produce the same warning.